src/test/java/io/zold/api/SmartTransactionTest.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.zold.api;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * Test case for {@link Transaction.Smart}.
+ *
+ * @author Izbassar Tolegen (t.izbassar@gmail.com)
+ * @version $Id$
+ * @since 0.1
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class SmartTransactionTest {
+
+    @Test
+    @Ignore
+    public void shouldFormatTransactionInProperFormat() {
+        MatcherAssert.assertThat(
+            // @checkstyle MagicNumberCheck (1 line)
+            new Transaction.Smart(1234L).toString(),
+            Matchers.startsWith(
+                // @checkstyle LineLengthCheck (1 line)
+                "3F;2018-07-03T10:15:30Z;4D2;pplIe28s;38ab8fc8e735c4fc;For programming;2sXoN5"
+            )
+        );
+    }
+
+    @Test
+    public void shouldReturnAmount() {
```

Comment:
@t-izbassar let's have tests that check whether `Transaction.Smart` amount enforces validity, i.e. within the bounds of a 16-bit unsigned int.